### ISSUE TITLE
Allow setting FFmpeg video parameters

### ISSUE THREAD

**Issue by veikk0** [alexheretic/ab-av1#10]
Currently, setting the GOP size with `-g` is essential when using SVT-AV1 via FFmpeg, because the default GOP size is fairly small and results in larger files than necessary. Another thing to consider is `-sc_detection 1` to enable scene change detection, which is currently off by default since it's not optimized. However, I still prefer it to having a constant GOP size.

There's also an `-svtav1-params` option [being planned](https://gitlab.com/AOMediaCodec/SVT-AV1/-/merge_requests/1830), but no code for this has been merged to either SVT-AV1 or FFmpeg currently.

While being able to set `-g` and `-sc_detection` would be good enough, being able to define any FFmpeg output options (including filters) would be ideal. But this does bring up extra complexity with VMAF calculation in situations that involve any modification of the source, like cropping or scaling, since such operations will also have to be applied to the source file when calculating VMAF.

In case it helps, I'll include a couple of sample FFmpeg commands I use for calculating the VMAF for modified encodes. I'm not sure if the setpts filter is required for this use case, but I think I found it necessary in the past for some x265 timestamp issues.

Scaled encode:

>ffmpeg -hide_banner -r 24 -i 720p.mkv -r 24 -i source.mp4 -an -sn -map 0:V -map 1:V -lavfi "[0:v]setpts=PTS-STARTPTS[dist];[1:v]setpts=PTS-STARTPTS,scale=1280:-2[ref];[dist][ref]libvmaf=n_threads=16" -f null -

60 fps source, 30 fps distorted file:

>ffmpeg -hide_banner -r 30 -i 30fps.x264.medium.RF23.mkv -r 60 -i ../bbb_sunflower_1080p_60fps_normal.mp4 -an -sn -map 0:V -map 1:V -lavfi "[0:v]setpts=PTS-STARTPTS[dist];[1:v]setpts=PTS-STARTPTS,fps=30[ref];[dist][ref]libvmaf=n_threads=16" -f null -

**Comment by alexheretic** [comment]
I was leaving towards just not supporting advanced ffmpeg usage and suggesting that be done directly. However, it may make sense to add fmpeg args. It could be a leaky abstraction though.

For example, we're not encoding av1 using ffmpeg. Ffmpeg simple converts to yuv and pipes into SvtAv1EncApp (a bit like I described https://alexheretic.github.io/posts/av1-p1/#re-encoding-h264--friends).

On the other hand I'm generally happy to expose any svt-av1 option on ab-av1 that someone has a user for. See https://gitlab.com/AOMediaCodec/SVT-AV1/-/blob/master/Docs/svt-av1_encoder_user_guide.md & let me know which ones you'd want exposed. I actually tried adding scene detection but found it had little effect on my test inputs so didn't commit.

**Comment by veikk0** [comment]
Oh, I didn't realize this uses SvtAv1EncApp directly.

In SvtAv1EncApp, maximum GOP size is set with `--keyint` and scene change detection is controlled with `--scd`. These two would be good to have.

>I actually tried adding scene detection but found it had little effect on my test inputs so didn't commit.

The maximum GOP size is so small by default that scene change detection won't have much of an effect. But with an allowed max GOP size of 10 seconds (which is pretty standard), the encoder has better chances to place keyframes in locations where there's a scene change while also maintaining efficiency.

Whether having support for FFmpeg filters is worth it does depend on the use case. Not being able to crop or scale would limit the program's usefulness as an all-in-one encoding tool (for example, I've got Blu-rays that I want to crop). crf-search would still be useful for non-scaling use cases, because cropping the black borders doesn't affect quality much (if I remember correctly). I can definitely see myself mostly using the auto-encode command.

Another thing I noticed is that the FFmpeg pipe uses the `yuv420p10le` pixel format by default, and this results in the encode being 10-bit. I know some people will say it makes no sense to use AV1 to encode anything but 10-bit, but I think there should be a choice for `yuv420p` too.
